File: iOS/AgoraEntScenarios/Common/API/CallAPI/CallAPI/Classes/SignalClient/CallBaseSignalClient.swift
//
//  CallBaseSignalClient.swift
//  CallAPI
//
//  Created by wushengtao on 2024/2/28.
//

import Foundation

@objcMembers open class CallBaseSignalClient: NSObject {
    public let delegates:NSHashTable<ISignalClientListener> = NSHashTable<ISignalClientListener>.weakObjects()
    
    public func addListener(listener: ISignalClientListener) {
        if delegates.contains(listener) { return }
        delegates.add(listener)
    }
    
    public func removeListener(listener: ISignalClientListener) {
        delegates.remove(listener)
    }
}

File: Android/common/scenario-api/lib_callapi/src/main/java/io/agora/onetoone/signalClient/CallBaseSignalClient.kt
package io.agora.onetoone.signalClient

open class CallBaseSignalClient {
    val listeners = mutableListOf<ISignalClientListener>()

    fun addListener(listener: ISignalClientListener) {
        if (listeners.contains(listener)) return
        listeners.add(listener)
    }

    fun removeListener(listener: ISignalClientListener) {
        listeners.add(listener)
    }
}